File: sdk/src/main/java/ru/usedesk/sdk/ui/knowledgebase/entity/DataOrMessage.java
package ru.usedesk.sdk.ui.knowledgebase.entity;

public class DataOrMessage<T> {
    private T data;
    private Message messageState;

    public DataOrMessage(T data) {
        this.data = data;
        this.messageState = Message.NONE;
    }

    public DataOrMessage(Message messageState) {
        this.messageState = messageState;
    }

    public T getData() {
        return data;
    }

    public Message getMessage() {
        return messageState;
    }

    public enum Message {
        LOADING, ERROR, NONE;
    }
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/entity/DataOrMessage.kt
package ru.usedesk.knowledgebase_gui.entity

class DataOrMessage<T> {
    var data: T? = null
        private set
    var message: Message
        private set

    constructor(data: T) {
        this.data = data
        message = Message.NONE
    }

    constructor(messageState: Message) {
        message = messageState
    }

    enum class Message {
        LOADING, ERROR, NONE
    }
}